src/entries/background/handlers/handleProviderRequest.ts

```diff
@@ -393,6 +398,57 @@ export const handleProviderRequest = ({
           }
           break;
         }
+        case 'wallet_watchAsset': {
+          const { featureFlags } = featureFlagsStore.getState();
+          if (!featureFlags.custom_rpc && process.env.IS_TESTING === 'false') {
+            throw new Error('Method not supported');
+          } else {
+            const {
+              type,
+              options: { address, symbol, decimals },
+            } = params as unknown as {
+              type: string;
+              options: {
+                address: Address;
+                symbol?: string;
+                decimals?: number;
+              };
+            };
+
+            if (type !== 'ERC20') {
+              throw new Error('Method supported only for ERC20');
+            }
+
+            if (!address) {
+              throw new Error('Address is required');
+            }
+
+            const activeSession = getActiveSession({ host });
+            let chainId = null;
+            if (activeSession) {
+              chainId = activeSession?.chainId;
+            } else {
+              chainId = deriveChainIdByHostname(host);
```

Comment:
we should probably block this request if there's no session because if you don't have a session on a network your trying to add a token manually means that you don't have the custom rpc added

this case will produce that `getProvider` calls you have below in the ui will crash the ui

Reply:
actually no, you could have the rpc with no session, but this is still a problem when the user doesn't have a session but the user also doesn't have the rpc

Reply:
i believe it defaults to mainnet now that i remember, so it won't crash but will show wrong info to the user?

Reply:
We discussed this on slack.  Dapps like etherscan have no session to suggest a token so we can't block it.
The user having the RPC or not is a separate thing.  A token can be suggested for mainnet or any chain we do support.  If we don't know the chain we default to mainnet and show a warning,  but we let the user choose the network from the list. There's a screen recording on the "screenshot" section of the PR.